fv3config/fv3run/_run.py

```diff
@@ -0,0 +1,32 @@
+import os
+from ._docker import _run_in_docker
+from ._native import _run_native
+
+
+def run(config_dict_or_location, outdir, runfile=None, docker_image=None, keyfile=None):
```

Comment:
It would be nice if there were a default docker image, but then you would need a different way to specify that the run should occur on docker.

Reply:
Hmm. I wonder if this top-level `run` function is a little too inflexible for all use-cases. For instance, there is currently no way to add flags to the docker command. I would suggest making `_run_native` and `_run_in_docker` public methods since those will be easier to extend.

Reply:
This run function is meant to replicate and be the only call for the command-line API, and I think most methods of running in docker or native are ones we want to expose in the command-line API. In other words, I think making `run_native` and `run_in_docker` public doesn't remove the need to extend `run` to support this behavior. And the code of `run` itself is easy to extend because it separates concerns by passing along to `_run_native` or `_run_in_docker` immediately.

Part of the zen of Python is that there should be ideally one obvious way of doing a thing. It's unclear to me what the user gains from having either e.g. both `run` or `run_native` to run the model natively. If it seems cleaner to have `run_docker` and `run_native` *instead* of `run`, and make `run` a private interface called by the command-line endpoint, that might be a nice route to go.

Reply:
Especially given I'm already finding I don't really want to add kubernetes into the `run` endpoint, because it adds *lots* of new options.